### DISCUSSION TITLE
How to train with part of the whole dataset?

### DISCUSSION THREAD

**Discussion by jarvishou829** [open-mmlab/mmdetection3d#2052] [Q&A]
Thanks for the great job. I wonder how to train with part of a whole big dataset like nuscenes. Because the trainval is too large and mini is too small. I wonder if I can spilt a quarter of the trainval dataset for training.

**Comment by ZwwWayne** [comment]
Hi @jarvishou829 ,
You can try MMDet3D 1.x and simply setting `indices=N` in the dataset can allow you to train the model only by N samples of the dataset.

**Reply by jarvishou829** [reply]
Thanks for your fast reply. Is the `indices` an arg of `mmdet.datasets.DATASETS`? I initialize the `NuScenesDataset` and register it to the `mmdet.datasets.DATASETS`, so should I set` indices `in the __init__ of `NuScenesDataset`?

**Reply by JingweiZhang12** [reply]
`indices` is an argument of `mmengine.BaseDataset`. https://github.com/open-mmlab/mmengine/blob/main/mmengine/dataset/base_dataset.py#L219

**Reply by jarvishou829** [reply]
That helps a lot! I will check and try it. Thanks.
